src/core/execute.c

```diff
@@ -964,7 +967,7 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                 prefix, empty_to_root(c->root_directory),
                 prefix, yes_no(c->root_ephemeral),
                 prefix, yes_no(c->non_blocking),
-                prefix, yes_no(c->private_tmp),
+                prefix, yes_no(c->private_tmp != PRIVATE_TMP_OFF),
```

Comment:
We should dump the actual mode of PrivateTmp= here, as this is for debug purpose?

Reply:
I think it's better to leave it yes/no, as it was requested here https://github.com/systemd/systemd/pull/32724#issuecomment-2104821696

Reply:
(but _dump are for debugging purposes only?)

Reply:
It still shows the values as configured

Reply:
i agree with @YHNdnzj, the dump should show the internal value, not the exported value

Reply:
I was implementing your comment as-is, but sure I'll change it